sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala

```diff
@@ -48,6 +48,20 @@ object MetalsPlugin extends AutoPlugin {
         )
     },
     javaSemanticdbEnabled := bspEnabled.value,
+    javaHome := {
+      javaHome.value match {
+        case None =>
+          // In case if jdk is >= 17 we need to set `javaHome` explicitly
```

Comment:
Why this is the case only when `jdk is >= 17`?
I took a quick look at `sbt/zinc` and it seems like the case for all java versions 🤔

Reply:
starting from jdk17 a lot of modules were closed and has to be explicitly opened, otherwise you'll get 
```shell
java.lang.IllegalAccessError: class com.sourcegraph.semanticdb_javac.SemanticdbPlugin 
(in unnamed module @0x29e5de87) cannot access class com.sun.tools.javac.api.BasicJavacTask 
(in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.api 
to unnamed module @0x29e5de87
```
like I got

Reply:
Ok, I got it!

- So, from JDK16, we have to pass flags like `add-exports` to access JDK internals as https://openjdk.java.net/jeps/396
- Metals pass `add-exports` flags via   https://github.com/scalameta/metals/blob/d4bf310e34eeb67539475a248d3a453549417413/sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala#L80-L88
  - As documented in https://sourcegraph.github.io/lsif-java/docs/getting-started.html#java
- But if we use `LocalJavaCompiler`, it doesn't accept those flags, and we hit `IllegalAccessError`

👍